record_api/core.py

```diff
@@ -383,11 +408,30 @@ def __call__(self) -> None:
                 (self.TOS, self.TOS1), BINARY_OPS[opname], (self.TOS1, self.TOS)
             )
 
+        if self.previous_stack and self.previous_stack.opname in FUNCTION_CALL_OP_NAMES:
+            self.log_called_method()
+
         method_name = f"op_{opname}"
         if hasattr(self, method_name):
             getattr(self, method_name)()
         return None
 
+    def log_called_method(self):
+        if self.previous_stack.log_call_args:
+            tos = self.TOS
+            if type(tos) is type and issubclass(tos, Exception):
+                # Don't record exception
+                return
+            filename, line, fn, args, *kwargs = self.previous_stack.log_call_args
+            kwargs = kwargs[0] if kwargs else {}
+            log_call(
+                f"{filename}:{line}",
+                fn,
+                tuple(args),
+                *((kwargs,) if kwargs else ()),
+                return_type=type(tos),
+            )
+
     # special case subscr b/c we only check first arg, not both
     def op_BINARY_SUBSCR(self):
         self.process((self.TOS1,), op.getitem, (self.TOS1, self.TOS))
```

Comment:
Can we record the return type for any of some these special methods as well?

Reply:
Yes, I think so.